### PR TITLE
add note to install.rst about how to run tests

### LINKED ISSUE #968: Clarifications for dependencies and unittests
Hi,

For a packager (like me :) ) it's useful to have full list of dependencies and also what we need to run tests (and how).

There is some discrepancies between https://github.com/blaze/dask#dependencies and http://dask.pydata.org/en/latest/install.html

For the unittests, it seems that we need python-pygraphviz and python-cachey

Thanks!

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #968

@sciunto what other information should we include here that would help someone in your position?

### CODE REVIEW

**Comment by sciunto** [comment]
Initially, I used nosetests and I got errros.

Here is my test about the unittests.
I made a virtualenv and cloned the repository. 

```
$ pip install pytest    
$ py.test dask    
 [...]  
ImportError: No module named 'partd'    
```

So, I guess partd must be installed. I discovered that toolz and pandas were missing. see https://github.com/blaze/partd/issues/5.

Even if partd is installed with pandas and toolz in my virtualenv, I still have import problems when I run py.test. However, python -c 'import partd' works. No clue right now...

**Comment by sciunto** [comment]
About python-pygraphviz and python-cachey, even if the tests are skipped if they are not installed, it's probably useful to indicate them as optional deps for running tests.

**Comment by mrocklin** [comment]
Where should we list these as optional dependencies, within the doc page or is there an appropriate location to put metadata like this?

**Comment by mrocklin** [comment]
Perhaps you would be satisfied if you installed with `pip install dask[complete]` rather than `pip install dask`?

**Comment by sciunto** [comment]
@mrocklin The doc or the README file are the two places I'm looking at. The doc is good.

I'll make further tests with pip.

File: docs/source/install.rst [hunk 1]
```diff
@@ -34,3 +38,22 @@ To install dask from source, clone the repository from `github
...
+    py.test dask
+
+Although please aware that installing dask naively may not install all
+requirements by default.  Please read the ``pip`` section above that discusses
+requirements.  You may choose to install the ``dask[complete]`` which includes
+all dependencies for all collections.  Alternatively you may choose to test
+only certain submodules depending on the libraries within your environment.
+For example to test only dask core and dask array we would run tests as
+follows::
+
```
Comment (sciunto) [inline docs/source/install.rst]: OK with that, but as a packager, I need to install everything with `python setup.py install`. So, I cannot rely on pip to install the dependence. Here, we need to list the modules needed to run the tests (numexpr, partd, ...)

Comment (sciunto) [inline docs/source/install.rst]: At the top, "installing from source" would probably deserve a section.

**Comment by sciunto** [comment]
Above, about the import problem, that was my fault (wrong py.test binary called in my virtualenv).

**Comment by sciunto** [comment]
For the unittest, it seems that the hook must be updated (from blaze/dask to dask/dask).

**Comment by mrocklin** [comment]
I've added a note about `pip install -e` and a pointer to look in `setup.py`.  I plan to merge this soon (releasing a minor update today.)

**Comment by sciunto** [comment]
+1.

I'll rework on the packaging after that. FYI, scikit-image is going to use dask as a requirement. It will probably push the distribution to package dask if they are already shipping scikit-image.

**Comment by mrocklin** [comment]
I appreciate your involvement here.  I'm happy to work to bring dask packaging and testing up to more stringent standards, especially as it gains more adoption from mature projects like `scikit-image`.

**Comment by mrocklin** [comment]
Question about wheels.  Is there a way to publish a universal wheel while still supporting the `extras_require` options within the setup.py file?

**Pull request merged**